src/hotspot/share/compiler/compileTask.hpp

```diff
@@ -127,8 +127,8 @@ class CompileTask : public CHeapObj<mtCompiler> {
   bool         is_complete() const               { return _is_complete; }
   bool         is_blocking() const               { return _is_blocking; }
   bool         is_success() const                { return _is_success; }
-  void         set_directive(const DirectiveSet* directive) { _directive = directive; }
-  const DirectiveSet* directive() const          { return _directive; }
+  void         set_directive(DirectiveSet* directive) { _directive = directive; }
+  DirectiveSet* directive()                      { return _directive; }
```

Comment:
The method can still be `const` since we are not changing `this`.
```suggestion
  DirectiveSet* directive() const                { return _directive; }
```